### ISSUE TITLE
HAL_attack_distance

### ISSUE THREAD

**Issue by SKYNETCyber** [Radekj1/NR6-HAL#19]
Major issue with attacking distance:
Foot infantry, ground vehicles and aircraft have a kind of limit in kilometers within which they can attack enemies. For example if commander have have infantry units uder its control and spotted enemy more than 5km away from closeset units, it will stop doing anything, commander will stop recon and capturing objectives unless you have RydHQ_RapidCapt set to 1000
In this example I used module Always Known Enemy (AlwaysKnownU), I synced it with commander and single enemy unit. 
Load scenario, open Zeus and delete (Provler and AH-99) car and armed helicopter (just for this test) As you can see, enemy is more than 5 kilometers away and for some reasons commander cannot attack that enemy. If you open Zeus and move enemy 1-2km close, commander will send units to attack it
Same issue with vehicles and armed helicopters. Restart mission, delete infantry and helicopter. Car is around 10km away from enemy and cann
![6](https://github.com/user-attachments/assets/7afb789f-d2e4-4bbd-83fc-b9cfec37195b)
![2](https://github.com/user-attachments/assets/8a8ebfef-4eec-4968-b862-3761a7bd9e89)
![3](https://github.com/user-attachments/assets/c2565838-cf4f-4f36-be52-08af086af93a)
![4](https://github.com/user-attachments/assets/6a40b82c-2f00-4443-b0cd-0bec2d0f17d1)
![5](https://github.com/user-attachments/assets/18f46fef-8c3c-4ddc-90ea-b4b83c52de7d)
ot attack it, move enemy 1-2km close and commander will send car to attack it.
Same test again but now delete infantry and car. Helicopter is more than 20km away and not attacking, move enemy a little close and commander will send it to attack
Thus, the limit for in
![1](https://github.com/user-attachments/assets/319e7a6f-17e9-4803-98a8-ba69b5a63311)
fantry is 5 km, for ground vehicles — 10 km, and for helicopters — 20 km. 
I do not know if this is a bug or was designed to be like this. 

In screenshot 1, an example with infantry that is further than 5 km, the commander simply skips cycles and cannot attack this enemy
In screenshot 2, I moved the enemy a little closer, and when the next cycle started, the commander began to attack this enemy.

Screenshots 3 and 4 show the same test, but with ground vehicles

Screenshots 5 and 6 show the same test, but with helicopter

**Comment by Radekj1** [comment]
I don't believe it is an issue, though for air / ground attack maybe I should increase the range, not sure in ArmA exactly, but human walking speed is around little less than 5km/h.
